kitsune/users/models.py

```diff
@@ -458,7 +464,72 @@ class AccountEvent(models.Model):
                                                   default=None,
                                                   null=True,
                                                   blank=False)
-    fxa_uid = models.CharField(blank=True, null=True, unique=True, max_length=128)
+    fxa_uid = models.CharField(blank=True, null=True, max_length=128)
     jwt_id = models.CharField(max_length=256)
     issued_at = models.CharField(max_length=32)
     profile = models.ForeignKey(Profile, related_name='account_events', null=True)
+    error = models.TextField(max_length=4096, blank=True)
+
+    def process(self):
```

Comment:
Personally, I would go for atomic tasks that handle each case. Eg this process method could just be something like:
```python
def process:
   if even_type == self.DELETE_USER:
        process_delete_task.apply_async() # or delay
   elif another_type_here:
         process_other_type_task.apply()
```

The above will keep this method short, easy to read and make it act as a kind of scheduler which is what we want here. Processing should take place in each task.